### ISSUE TITLE
Build hangs on property file (v 1.5.0, pg11b2)

### ISSUE THREAD

**Issue by fluca1978** [tada/pljava#160]
When building pljava I see the system hangs on loading the properties file, and after 20+ minutes nothing happened:
```
% mvn -Dnar.cores=1 -X -Pwnosign   clean install
...
[DEBUG] Configuring mojo org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2:read-project-properties from plugin realm ClassRealm[plugin>org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2, parent: sun.misc.Launcher$AppClassLoader@70dea4e]
[DEBUG] Configuring mojo 'org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2:read-project-properties' with basic configurator -->
[DEBUG]   (f) files = [/usr/home/luca/pljava/pljava-so/target/pgsql.properties]
[DEBUG]   (f) project = MavenProject: org.postgresql:pljava-so:1.5.0 @ /usr/home/luca/pljava/pljava-so/pom.xml
[DEBUG]   (f) quiet = false
[DEBUG] -- end configuration --
[DEBUG] Loading property file: /usr/home/luca/pljava/pljava-so/target/pgsql.properties
```

Here's information about the system:
```
% gcc --version    
FreeBSD clang version 4.0.0 (tags/RELEASE_400/final 297347) (based on LLVM 4.0.0)
Target: x86_64-unknown-freebsd11.1
Thread model: posix
InstalledDir: /usr/bin
% openssl version
OpenSSL 1.0.2k-freebsd  26 Jan 2017
```

the postgresql version is 11 beta 2.
The pljava version is 1.5.0: ```% git checkout refs/tags/V1_5_0```.

**Comment by fluca1978** [comment]
The same happens with `HEAD`.
In the case it matters, here's the property file content:
```
% cat /usr/home/luca/pljava/pljava-so/target/pgsql.properties
#Mon Jul 23 14:22:32 CEST 2018
PGSQL_BINDIR=/opt/pg11b2/bin
PGSQL_INCLUDEDIR-SERVER=/opt/pg11b2/include/postgresql/server
PGSQL_LIBDIR=/opt/pg11b2/lib
PGSQL_VER_CLASSIFIER=${PGSQL_VER_CLASSIFIER}
PGSQL_INCLUDEDIR=/opt/pg11b2/include
PGSQL_PKGLIBDIR=/opt/pg11b2/lib/postgresql
MSVC_RINT=msvc.rint
```

**Comment by jcflack** [comment]
I don't have an immediate explanation for the hang. It is not something I have seen before.

However, 1.5.0 is too old for what you are attempting; only 1.5.1 (beta1) has the necessary updates to build with PG 9.6 or 10. No testing with PG 11 has been done yet (you'd be the first, as far as I know). Looking through PG 11 release notes, I believe that some adjustments will be necessary..

When you built from HEAD, was that HEAD of `master`, or HEAD of `REL1_5_STABLE`? At the moment, `REL1_5_STABLE` is the focus of development; `master` will eventually become a future major release, but will have to be brought up to date by merging `REL1_5_STABLE` into it after the release of 1.5.1.

**Comment by fluca1978** [comment]
I tried to rebuild with 1_5_1-STABLE but got the same property file hanging:

```
% git checkout refs/tags/REL1_5_STABLE-BASE
% mvn -Dnar.cores=1  -Pwnosign  -e -X clean install
...
Setting project property: PGSQL_VER -> PostgreSQL 11beta2
[loadresource] loading PostgreSQL 11beta2 into property PGSQL_VER_CLASSIFIER
[loadresource] resource size = 18
[loadresource] loading /opt/pg11b2/include/pg_config.h into property KNOWS_MSVC_RINT
[loadresource] resource size = 31872
Overriding previous definition of property "MSVC_RINT"
Setting project property: MSVC_RINT -> msvc.rint
Property "PGSQL_VER_CLASSIFIER" has not been set
[propertyfile] Creating new property file: /usr/home/luca/pljava/pljava-so/target/pgsql.properties
      [ant] Exiting /usr/home/luca/pljava/pljava-so/build.xml.
[INFO] Executed tasks
[INFO] 
[INFO] --- properties-maven-plugin:1.0-alpha-2:read-project-properties (default) @ pljava-so ---
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=39882, ConflictMarker.markTime=46100, ConflictMarker.nodeCount=31, ConflictIdSorter.graphTime=13320, ConflictIdSorter.topsortTime=12949, ConflictIdSorter.conflictIdCount=14, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=130098, ConflictResolver.conflictItemCount=31, DefaultDependencyCollector.collectTime=1777946, DefaultDependencyCollector.transformTime=259803}
[DEBUG] org.codehaus.mojo:properties-maven-plugin:jar:1.0-alpha-2:
[DEBUG]    org.apache.maven:maven-model:jar:2.0.6:compile
[DEBUG]    org.apache.maven:maven-plugin-api:jar:2.0.6:compile
[DEBUG]    org.apache.maven:maven-project:jar:2.0.6:compile
[DEBUG]       org.apache.maven:maven-settings:jar:2.0.6:compile
[DEBUG]       org.apache.maven:maven-profile:jar:2.0.6:compile
[DEBUG]       org.apache.maven:maven-artifact-manager:jar:2.0.6:compile
[DEBUG]          org.apache.maven:maven-repository-metadata:jar:2.0.6:compile
[DEBUG]       org.apache.maven:maven-plugin-registry:jar:2.0.6:compile
[DEBUG]       org.apache.maven:maven-artifact:jar:2.0.6:compile
[DEBUG]       org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:compile
[DEBUG]          junit:junit:jar:3.8.2:test
[DEBUG]          classworlds:classworlds:jar:1.1-alpha-2:compile
[DEBUG]    org.codehaus.plexus:plexus-utils:jar:1.5.6:compile
[DEBUG] Using transporter WagonTransporter with priority -1.0 for https://repo.maven.apache.org/maven2
[DEBUG] Using connector BasicRepositoryConnector with priority 0.0 for https://repo.maven.apache.org/maven2
Downloading from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/1.5.6/plexus-utils-1.5.6.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/1.5.6/plexus-utils-1.5.6.jar (251 kB at 665 kB/s)
[DEBUG] Writing tracking file /home/luca/.m2/repository/org/codehaus/plexus/plexus-utils/1.5.6/_remote.repositories
[DEBUG] Writing tracking file /home/luca/.m2/repository/org/codehaus/plexus/plexus-utils/1.5.6/plexus-utils-1.5.6.jar.lastUpdated
[DEBUG] Created new class realm plugin>org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2
[DEBUG] Importing foreign packages into class realm plugin>org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2
[DEBUG]   Imported:  < project>org.postgresql:pljava-so:1.5.0-SNAPSHOT
[DEBUG] Populating class realm plugin>org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2
[DEBUG]   Included: org.codehaus.mojo:properties-maven-plugin:jar:1.0-alpha-2
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:1.5.6
[DEBUG] Configuring mojo org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2:read-project-properties from plugin realm ClassRealm[plugin>org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2, parent: sun.misc.Launcher$AppClassLoader@70dea4e]
[DEBUG] Configuring mojo 'org.codehaus.mojo:properties-maven-plugin:1.0-alpha-2:read-project-properties' with basic configurator -->
[DEBUG]   (f) files = [/usr/home/luca/pljava/pljava-so/target/pgsql.properties]
[DEBUG]   (f) project = MavenProject: org.postgresql:pljava-so:1.5.0-SNAPSHOT @ /usr/home/luca/pljava/pljava-so/pom.xml
[DEBUG]   (f) quiet = false
[DEBUG] -- end configuration --
[DEBUG] Loading property file: /usr/home/luca/pljava/pljava-so/target/pgsql.properties
```

and the properties file is
```
% cat /usr/home/luca/pljava/pljava-so/target/pgsql.properties
#Mon Jul 23 17:25:08 CEST 2018
PGSQL_BINDIR=/opt/pg11b2/bin
PGSQL_INCLUDEDIR-SERVER=/opt/pg11b2/include/postgresql/server
PGSQL_LIBDIR=/opt/pg11b2/lib
PGSQL_VER_CLASSIFIER=${PGSQL_VER_CLASSIFIER}
PGSQL_INCLUDEDIR=/opt/pg11b2/include
PGSQL_PKGLIBDIR=/opt/pg11b2/lib/postgresql
MSVC_RINT=msvc.rint
```


**The same happens with pg11b1 and pg 10.4, therefore it seems a problem of my system, but I don't know what to search for.**

**Comment by jcflack** [comment]
I wonder why `PGSQL_VER_CLASSIFIER` is not getting set. Perhaps there is something about the PG 11 version string that does not match the expected pattern? As long as you are building 1.5.1-BETA1, the pattern has already been adjusted for the version numbering change in PG 10, so I am puzzled how anything could go wrong with 11. Does your PG build have some extra information added into the version string? (The pattern also tries to accommodate that, but maybe it is possible for some extra version info to fail to match.)

I am still not sure how that would lead to _hanging_ when loading the property file, unless the parser for that file treats `PGSQL_VER_CLASSIFIER=${PGSQL_VER_CLASSIFIER}` as an infinite expansion....

**Comment by fluca1978** [comment]
This is the expansion part:
```
Setting project property: PGSQL_VER -> PostgreSQL 10.4
[loadresource] loading PostgreSQL 10.4 into property PGSQL_VER_CLASSIFIER
[loadresource] resource size = 15
```

However, even with 10.4 I got the following:

```
% pg_config --version
PostgreSQL 10.4
```

while in the case of the beta version
```
% pg_config --version
PostgreSQL 11beta2
```

**Comment by fluca1978** [comment]
Uhm...after upgrading to gcc 7, it seems a little step further has been done and the properties file seems correct so far:

```
% gcc --version
gcc (FreeBSD Ports Collection) 7.3.0

% cat ./pljava-so/target/pgsql.properties
#Mon Jul 23 18:58:46 CEST 2018
PGSQL_BINDIR=/opt/pg11b2/bin
PGSQL_INCLUDEDIR-SERVER=/opt/pg11b2/include/postgresql/server
PGSQL_LIBDIR=/opt/pg11b2/lib
PGSQL_VER_CLASSIFIER=pg11.99
PGSQL_INCLUDEDIR=/opt/pg11b2/include
PGSQL_PKGLIBDIR=/opt/pg11b2/lib/postgresql
MSVC_RINT=msvc.rint
```

but if I try to build against master, I got the same hanging and the properties file does not catch up the version string well (so, the hanging seems really an infinite loop against it:
```
% cat /usr/home/luca/pljava/pljava-so/target/pgsql.properties
#Mon Jul 23 19:27:16 CEST 2018
PGSQL_BINDIR=/opt/pg11b2/bin
PGSQL_INCLUDEDIR-SERVER=/opt/pg11b2/include/postgresql/server
PGSQL_LIBDIR=/opt/pg11b2/lib
PGSQL_VER_CLASSIFIER=${PGSQL_VER_CLASSIFIER}
PGSQL_INCLUDEDIR=/opt/pg11b2/include
PGSQL_PKGLIBDIR=/opt/pg11b2/lib/postgresql
MSVC_RINT=msvc.rint
```

So, getting back to 1.5.1b1, the4 builds now fails on nar:

```
% mvn -Dnar.cores=1  -Pwnosign -X -e clean install
...
[DEBUG] Execute:Java13CommandLauncher: Executing 'gcc' with arguments:
'-Wno-sign-conversion'
'-Wall'
'-Wno-long-long'
'-Wpointer-arith'
'-Wconversion'
'-fPIC'
'-c'
'-fPIC'
'-DPLJAVA_SO_VERSION=1.5.1-BETA1'
'-DFreeBSD'
'-DGNU_GCC'
'-I/usr/home/luca/pljava/pljava-so/target/nar/javah-include'
'-I/usr/local/openjdk8/include'
'-I/usr/local/openjdk8/include/freebsd'
'-I/opt/pg11b2/include'
'-I/opt/pg11b2/include/postgresql/server'
'-I/usr/home/luca/pljava/pljava-so/src/main/include'
'-I/usr/home/luca/pljava/pljava-so/target/nar/javah-include'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/AclId.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Any.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Array.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Backend.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/BigDecimal.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Boolean.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Byte.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Coerce.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Composite.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Date.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Double.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/ErrorData.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Exception.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/ExecutionPlan.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Float.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Function.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/HashMap.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/HeapTupleHeader.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/InstallHelper.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Integer.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Invocation.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Iterator.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/JNICalls.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/JavaWrapper.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Long.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Oid.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/PgObject.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/PgSavepoint.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Portal.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Relation.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SPI.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLInputFromChunk.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLInputFromTuple.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLOutputToChunk.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SQLOutputToTuple.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/Session.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Short.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/String.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/SubXactListener.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Time.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Timestamp.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/TriggerData.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Tuple.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/TupleDesc.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/TupleTable.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Type.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/UDT.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/Void.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c'
'/usr/home/luca/pljava/pljava-so/src/main/c/type/byte_array.c'

The ' characters around the executable and arguments are
not part of the command.
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/JavaWrapper.c: In function 'JavaWrapper_initialize':
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/JavaWrapper.c:66:35: error: macro "AllocSetContextCreate" passed 5 arguments, but takes just 3
[ERROR]            ALLOCSET_DEFAULT_MAXSIZE);
[ERROR]                                    ^
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/JavaWrapper.c:62:22: error: 'AllocSetContextCreate' undeclared (first use in this function)
[ERROR]   JavaMemoryContext = AllocSetContextCreate(TopMemoryContext,
[ERROR]                       ^~~~~~~~~~~~~~~~~~~~~
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/JavaWrapper.c:62:22: note: each undeclared identifier is reported only once for each function it appears in
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/SubXactListener.c: In function 'subXactCB':
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/SubXactListener.c:26:2: warning: enumeration value 'SUBXACT_EVENT_PRE_COMMIT_SUB' not handled in switch [-Wswitch]
[ERROR]   switch(event)
[ERROR]   ^~~~~~
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/Type.c: In function 'Type_invokeSRF':
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/Type.c:407:35: error: macro "AllocSetContextCreate" passed 5 arguments, but takes just 3
[ERROR]            ALLOCSET_DEFAULT_MAXSIZE);
[ERROR]                                    ^
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/Type.c:403:25: error: 'AllocSetContextCreate' undeclared (first use in this function)
[ERROR]    ctxData->rowContext = AllocSetContextCreate(context->multi_call_memory_ctx,
[ERROR]                          ^~~~~~~~~~~~~~~~~~~~~
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/type/Type.c:403:25: note: each undeclared identifier is reported only once for each function it appears in
[ERROR] In file included from /usr/home/luca/pljava/pljava-so/src/main/c/type/UDT.c:13:0:
[ERROR] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h: In function 'pq_writeint8':
[WARNING] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h:56:14: warning: conversion to 'int' from 'long unsigned int' may alter its value [-Wconversion]
[ERROR]   buf->len += sizeof(int8);
[ERROR]               ^~~~~~
[ERROR] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h: In function 'pq_writeint16':
[WARNING] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h:70:14: warning: conversion to 'int' from 'long unsigned int' may alter its value [-Wconversion]
[ERROR]   buf->len += sizeof(int16);
[ERROR]               ^~~~~~
[ERROR] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h: In function 'pq_writeint32':
[WARNING] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h:84:14: warning: conversion to 'int' from 'long unsigned int' may alter its value [-Wconversion]
[ERROR]   buf->len += sizeof(int32);
[ERROR]               ^~~~~~
[ERROR] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h: In function 'pq_writeint64':
[WARNING] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h:98:14: warning: conversion to 'int' from 'long unsigned int' may alter its value [-Wconversion]
[ERROR]   buf->len += sizeof(int64);
[ERROR]               ^~~~~~
[ERROR] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h: In function 'pq_writestring':
[WARNING] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h:114:15: warning: conversion to 'int' from 'size_t {aka long unsigned int}' may alter its value [-Wconversion]
[ERROR]   int   slen = strlen(str);
[ERROR]                ^~~~~~
[WARNING] /opt/pg11b2/include/postgresql/server/libpq/pqformat.h:119:10: warning: conversion to 'int' from 'size_t {aka long unsigned int}' may alter its value [-Wconversion]
[ERROR]    slen = strlen(p);
[ERROR]           ^~~~~~
[ERROR] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c: In function 'xactCB':
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PARALLEL_COMMIT' not handled in switch [-Wswitch]
[ERROR]   switch(event)
[ERROR]   ^~~~~~
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PARALLEL_ABORT' not handled in switch [-Wswitch]
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PRE_COMMIT' not handled in switch [-Wswitch]
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PARALLEL_PRE_COMMIT' not handled in switch [-Wswitch]
[WARNING] /usr/home/luca/pljava/pljava-so/src/main/c/XactListener.c:25:2: warning: enumeration value 'XACT_EVENT_PRE_PREPARE' not handled in switch [-Wswitch]
[INFO] 50 files were compiled.                                      
[INFO] 50 files were compiled.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] PostgreSQL PL/Java 1.5.1-BETA1 ..................... SUCCESS [  5.513 s]
[INFO] PL/Java API ........................................ SUCCESS [  8.827 s]
[INFO] PL/Java backend Java code .......................... SUCCESS [  7.803 s]
[INFO] PL/Java backend native code ........................ FAILURE [ 28.011 s]
[INFO] PL/Java Deploy ..................................... SKIPPED
[INFO] PL/Java Ant tasks .................................. SKIPPED
[INFO] PL/Java examples ................................... SKIPPED
[INFO] PL/Java packaging 1.5.1-BETA1 ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 52.553 s
[INFO] Finished at: 2018-07-23T19:30:19+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.maven-nar:nar-maven-plugin:3.2.3:nar-compile (default-nar-compile) on project pljava-so: NAR: Compile failed: gcc failed with return code 1 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.maven-nar:nar-maven-plugin:3.2.3:nar-compile (default-nar-compile) on project pljava-so: NAR: Compile failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.maven.plugin.MojoExecutionException: NAR: Compile failed
    at com.github.maven_nar.NarCompileMojo.createLibrary (NarCompileMojo.java:452)
    at com.github.maven_nar.NarCompileMojo.narExecute (NarCompileMojo.java:122)
    at com.github.maven_nar.AbstractNarMojo.execute (AbstractNarMojo.java:302)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:954)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.tools.ant.BuildException: gcc failed with return code 1
    at com.github.maven_nar.cpptasks.compiler.CommandLineCompiler.compile (CommandLineCompiler.java:222)
    at com.github.maven_nar.cpptasks.compiler.CommandLineCompilerConfiguration.compile (CommandLineCompilerConfiguration.java:148)
    at com.github.maven_nar.cpptasks.CCTask$Core.run (CCTask.java:946)
```

Seems really hard to get it done. Any idea?

**Comment by fluca1978** [comment]
gcc fails because from 10.4 to 11 beta the memutils.h has changed this marco:

```
extern MemoryContext AllocSetContextCreate(MemoryContext parent,
                                          const char *name,
                                          Size minContextSize,
                                          Size initBlockSize,
                                          Size maxBlockSize);
```

into this

```
/* aset.c */
extern MemoryContext AllocSetContextCreateExtended(MemoryContext parent,
                                                          const char *name,
                                                          Size minContextSize,
                                                          Size initBlockSize,
                                                          Size maxBlockSize);

/*
 * This wrapper macro exists to check for non-constant strings used as context
 * names; that's no longer supported.  (Use MemoryContextSetIdentifier if you
 * want to provide a variable identifier.)  Note you must specify block sizes
 * with one of the abstraction macros below.
 */
#ifdef HAVE__BUILTIN_CONSTANT_P
#define AllocSetContextCreate(parent, name, allocparams) \
        (StaticAssertExpr(__builtin_constant_p(name), \
                                          "memory context names must be constant strings"), \
         AllocSetContextCreateExtended(parent, name, allocparams))
#else
#define AllocSetContextCreate(parent, name, allocparams) \
        AllocSetContextCreateExtended(parent, name, allocparams)
#endif

```

**I'm not sure, but doing the following changes made the compilation to work:**

```
diff --git a/pljava-so/src/main/c/type/JavaWrapper.c b/pljava-so/src/main/c/type/JavaWrapper.c
index 2b0c4358..33a5f0fe 100644
--- a/pljava-so/src/main/c/type/JavaWrapper.c
+++ b/pljava-so/src/main/c/type/JavaWrapper.c
@@ -59,11 +59,11 @@ void JavaWrapper_initialize(void)
        PgObject_registerNatives2(s_JavaWrapper_class, methods);
        s_JavaWrapper_m_pointer = PgObject_getJavaField(s_JavaWrapper_class, "m_pointer", "J");
 
-       JavaMemoryContext = AllocSetContextCreate(TopMemoryContext,
-                                                                        "PL/Java",
-                                                                        ALLOCSET_DEFAULT_MINSIZE,
-                                                                        ALLOCSET_DEFAULT_INITSIZE,
-                                                                        ALLOCSET_DEFAULT_MAXSIZE);
+       JavaMemoryContext = AllocSetContextCreateExtended(TopMemoryContext,
+                                                 "PL/Java",
+                                                 ALLOCSET_DEFAULT_MINSIZE,
+                                                 ALLOCSET_DEFAULT_INITSIZE,
+                                                 ALLOCSET_DEFAULT_MAXSIZE);
 }
 
 /*
diff --git a/pljava-so/src/main/c/type/Type.c b/pljava-so/src/main/c/type/Type.c
index 9b12fa3f..2135c52a 100644
--- a/pljava-so/src/main/c/type/Type.c
+++ b/pljava-so/src/main/c/type/Type.c
@@ -400,7 +400,7 @@ Datum Type_invokeSRF(Type self, jclass cls, jmethodID method, jvalue* args, PG_F
                else
                        ctxData->spiContext = 0;
 
-               ctxData->rowContext = AllocSetContextCreate(context->multi_call_memory_ctx,
+               ctxData->rowContext = AllocSetContextCreateExtended(context->multi_call_memory_ctx,
                                                                  "PL/Java row context",
                                                                  ALLOCSET_DEFAULT_MINSIZE,
                                                                  ALLOCSET_DEFAULT_INITSIZE,
```

I've not tested yet if it runs...

**Comment by fluca1978** [comment]
No, seems it is not working:

```
testdb=# CREATE EXTENSION pljava;
FATAL:  extensions cannot define GUC_LIST_QUOTE variables
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Succeeded.
testdb=# select version();
                                                                         version                                                                         
---------------------------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 11beta2 on x86_64-unknown-freebsd11.1, compiled by FreeBSD clang version 4.0.0 (tags/RELEASE_400/final 297347) (based on LLVM 4.0.0), 64-bit
```

**Comment by fluca1978** [comment]
Seems that with this diff I get running pljava on 11 b2 and the examples run so far:

```
diff --git a/pljava-so/src/main/c/Backend.c b/pljava-so/src/main/c/Backend.c
index 8113909f..67ae79c3 100644
--- a/pljava-so/src/main/c/Backend.c
+++ b/pljava-so/src/main/c/Backend.c
@@ -1457,7 +1457,7 @@ static void registerGUCOptions(void)
                #endif
                PGC_USERSET,
                #if PG_VERSION_NUM >= 80400
-                       GUC_LIST_INPUT | GUC_LIST_QUOTE,
+                       GUC_LIST_INPUT,
                #endif
                #if PG_VERSION_NUM >= 90100
                        NULL, /* check hook */
diff --git a/pljava-so/src/main/c/type/JavaWrapper.c b/pljava-so/src/main/c/type/JavaWrapper.c
index 2b0c4358..33a5f0fe 100644
--- a/pljava-so/src/main/c/type/JavaWrapper.c
+++ b/pljava-so/src/main/c/type/JavaWrapper.c
@@ -59,11 +59,11 @@ void JavaWrapper_initialize(void)
        PgObject_registerNatives2(s_JavaWrapper_class, methods);
        s_JavaWrapper_m_pointer = PgObject_getJavaField(s_JavaWrapper_class, "m_pointer", "J");
 
-       JavaMemoryContext = AllocSetContextCreate(TopMemoryContext,
-                                                                        "PL/Java",
-                                                                        ALLOCSET_DEFAULT_MINSIZE,
-                                                                        ALLOCSET_DEFAULT_INITSIZE,
-                                                                        ALLOCSET_DEFAULT_MAXSIZE);
+       JavaMemoryContext = AllocSetContextCreateExtended(TopMemoryContext,
+                                                 "PL/Java",
+                                                 ALLOCSET_DEFAULT_MINSIZE,
+                                                 ALLOCSET_DEFAULT_INITSIZE,
+                                                 ALLOCSET_DEFAULT_MAXSIZE);
 }
 
 /*
diff --git a/pljava-so/src/main/c/type/Type.c b/pljava-so/src/main/c/type/Type.c
index 9b12fa3f..2135c52a 100644
--- a/pljava-so/src/main/c/type/Type.c
+++ b/pljava-so/src/main/c/type/Type.c
@@ -400,7 +400,7 @@ Datum Type_invokeSRF(Type self, jclass cls, jmethodID method, jvalue* args, PG_F
                else
                        ctxData->spiContext = 0;
 
-               ctxData->rowContext = AllocSetContextCreate(context->multi_call_memory_ctx,
+               ctxData->rowContext = AllocSetContextCreateExtended(context->multi_call_memory_ctx,
                                                                  "PL/Java row context",
                                                                  ALLOCSET_DEFAULT_MINSIZE,
                                                                  ALLOCSET_DEFAULT_INITSIZE,

```

on the database side:

```
# SET pljava.libjvm_location to '/usr/local/openjdk7/jre/lib/amd64/server/libjvm.so';
# alter database testdb set pljava.libjvm_location from current;
# SELECT sqlj.install_jar(
  'file:'
  '/buildroot/pljava-examples/target/pljava-examples-1.5.1-BETA1.jar',
  'examples', true);
# SELECT sqlj.set_classpath('javatest', 'examples');
# SELECT javatest.java_addone(3);
 java_addone 
-------------
           4

```

hope this helps (and keep working).

**Comment by jcflack** [comment]
> Uhm...after upgrading to gcc 7, it seems a little step further has been done and the properties file seems correct so far:

I cannot fathom how the gcc version affects the parsing of the PG version classifier. `pg11.99` is the string I would expect though, so that much worked.

> extern MemoryContext AllocSetContextCreate(...
> extern MemoryContext AllocSetContextCreateExtended(...

Thank you for finding that.

> FATAL:  extensions cannot define GUC_LIST_QUOTE variables

That was the exact thing I saw in the PG 11 release notes and figured was going to be trouble. Somewhere in the PG core discussion of that, there was a comment like "we never intended extensions to be able to do that anyway", so I'll clearly have to accommodate it in PL/Java with a backward-incompatible change to how that variable is defined.

**Comment by fluca1978** [comment]
At the moment I'm able to run my short examples, so I mark this issue as closed. I will report back any other changes in another issue if required.
